src/zenml/orchestrators/cache_utils.py

```diff
@@ -31,6 +31,30 @@
 logger = get_logger(__name__)
 
 
+def is_cache_enabled(
+    step_enable_cache: Optional[bool],
+    pipeline_enable_cache: Optional[bool],
+) -> bool:
+    """Checks if caching is enabled for a step run.
+
+    This is the case if:
+    - caching is explicitly enabled for the step, or
+    - caching is neither explicitly disabled for the step nor the pipeline.
+
+    Args:
+        step_enable_cache: The enable cache parameter of the step.
+        pipeline_enable_cache: The enable cache parameter of the pipeline.
+
+    Returns:
+        True if caching is enabled, False otherwise.
+    """
+    if step_enable_cache is not None:
+        return step_enable_cache
+    if pipeline_enable_cache is not None:
+        return pipeline_enable_cache
+    return True
```

Comment:
```suggestion
    return step_enable_cache or pipeline_enable_cache or True
```

Reply:
That's certainly not equivalent since `x or y or True` is always `True` :D